client/dev/sheet/sheet.jsx

```diff
@@ -0,0 +1,161 @@
+import React from "react";
+import AbilityScore from "./abilities.jsx";
+import Quirks from "./priorsandparticulars.jsx";
+
+var nextQuirksId = 1;
+
+var Sheet = React.createClass({
+  /* Set Properties */
+  propTypes: {
+    character: React.PropTypes.shape({
+      abilities: React.PropTypes.array,
+      quirks: React.PropTypes.array
+    })
+  },
+  
+  getDefaultProps: function() {
+    return {
+      initialStats: {
+        abilities: [
+          {
+            name: "Strength",
+            score: 1,
+            fraction: 0,
+            id: 1
+          },
+          {
+            name: "Intelligence",
+            score: 1,
+            fraction: 0,
+            id: 2            
+          },
+          {
+            name: "Wisdom",
+            score: 1,
+            fraction: 0,
+            id: 3
+          },
+          {
+            name: "Dexterity",
+            score: 1,
+            fraction: 0,
+            id: 4
+          },
+          {
+            name: "Constitution",
+            score: 1,
+            fraction: 0,
+            id: 5
+          },
+          {
+            name: "Looks",
+            score: 1,
+            fraction: 0,
+            id: 6
+          },
+          {
+            name: "Charisma",
+            score: 1,
+            fraction: 0,
+            id: 7
+          },
+          
+        ],
+        quirks: [
+        { 
+          name: "Test",
+          id: 0
+        }
+        ]
+      }
+    }
+  },
+  
+  /* Set state */  
+  getInitialState: function() {
+    return {
+      character: { 
+        stats: this.props.initialStats
+      }
+    };
+  },
+  
+  /* Define functions */ 
+  onScoreChange: function( index, delta ){
+    var newScore = this.state.character.stats.abilities[index].score + delta;
+    if( newScore <= 25 && newScore >= 1 ){
+      this.state.character.stats.abilities[index].score = newScore;
```

Comment:
It is bad practice to mutate state directly like this. Instead, create a new object, extend it with the existing state, and then apply your changes to that new object. Then set the state to the new object.

This will allow react to shallowly compare the old state object and new state object and know that they are different.

```javascript
var newState = Object.assign({}, this.state);
newState.character.stats.abilities[index].score = newScore;
this.setState(newState);
```